### PR TITLE
go_watch_it - added stop words and tests

### LINKED ISSUE #2070: GoWatchIt: Should ignore the query "pocket watch"
Unfortunately the query "pocket watch" triggers this IA and get's past our relevancy checking. 

https://duckduckgo.com/?q=pocket+watch&t=ffab&ia=watch

We may need to adjust the relevancy checking threshold to consider all words with 3+ characters, but it may be best to simply check if the query matches this term and if so, `return;`

------
IA Page:  http://duck.co/ia/view/go_watch_it

### PR DESCRIPTION
**Pull request by iambibhas**
Fixes #2070.

 - Added list of stop words
 - added tests for them

### CODE REVIEW

File: lib/DDG/Spice/GoWatchIt.pm [hunk 1]
```diff
@@ -30,9 +32,10 @@ spice wrap_jsonp_callback => 1;
 handle remainder_lc => sub {
 
   return unless $_; # Guard against "no answer"
-  $_ =~ s/$killwords//g; # remove killwords
-  return trim($_); # trim spaces and return
+  return if $_ =~ m/^$stopwords$/g; # Guard against stop words "apple watch"
```
Comment (moollaza) [inline lib/DDG/Spice/GoWatchIt.pm]: I think we should compare the full query against these as the order matters: "apple watch" vs "watch apple".

We expose the full query in the `$req` object.

File: lib/DDG/Spice/GoWatchIt.pm [hunk 2]
```diff
@@ -30,9 +32,10 @@ spice wrap_jsonp_callback => 1;
...
 
+  $_ =~ s/$ignorewords//g; # remove ignorewords
```
Comment (moollaza) [inline lib/DDG/Spice/GoWatchIt.pm]: this needs `\b` around the regex so we don't match a substring.

File: lib/DDG/Spice/GoWatchIt.pm (hunk 1 above)
Comment (moollaza) [inline lib/DDG/Spice/GoWatchIt.pm]: Also need `\b` here

**Comment by moollaza** [comment]
Ping! @iambibhas just checking in here if you're still interested in finishing this up?

**Comment by iambibhas** [comment]
@moollaza I'll make the changes and send a PR tonight(in 6-8 hours).

**Comment by moollaza** [comment]
LGTM :+1:

**Comment by moollaza** [comment]
@iambibhas thanks a ton!

**Pull request merged**